src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts

```diff
@@ -381,8 +391,11 @@ export class InlineChatWidget {
 		let resultingAppender: IInlineChatMessageAppender | undefined;
 		if (hasMessage) {
 			const sessionModel = this._chatMessageDisposables.add(new ChatModel(message.providerId, undefined, this._logService, this._chatAgentService, this._instantiationService));
+			this._sessionId = sessionModel.sessionId;
 			const responseModel = this._chatMessageDisposables.add(new ChatResponseModel(message.message, sessionModel, undefined, undefined, message.requestId, !isIncomplete, false, undefined));
-			const viewModel = this._chatMessageDisposables.add(new ChatResponseViewModel(responseModel, this._logService));
+			this._responseViewModel = this._chatMessageDisposables.add(new ChatResponseViewModel(responseModel, this._logService));
+			const chatViewModel = this._instantiationService.createInstance(ChatViewModel, sessionModel, this._codeBlockModelCollection);
+			chatViewModel.updateCodeBlockTextModels(this._responseViewModel);
```

Comment:
Since the `ChatViewModel` is created as progress/events are occurring, the listeners to such events are registered after the events have already happened. Force update here so that `_codeBlockModelCollection` gets populated.